pre_commit_hooks/forbid_new_submodules.py

```diff
@@ -1,14 +1,26 @@
-from typing import Optional
+import os
+import sys
 from typing import Sequence
 
 from pre_commit_hooks.util import cmd_output
 
 
-def main(argv: Optional[Sequence[str]] = None) -> int:
-    # `argv` is ignored, pre-commit will send us a list of files that we
-    # don't care about
+def main(argv: Sequence[str] = sys.argv[1:]) -> int:
+    if not argv:
+        return 0
+    if (
+        'PRE_COMMIT_FROM_REF' in os.environ and
+        'PRE_COMMIT_TO_REF' in os.environ
+    ):
+        diff_args: Sequence[str] = (
+            '--merge-base',
+            os.environ['PRE_COMMIT_FROM_REF'],
+            os.environ['PRE_COMMIT_TO_REF'],
+        )
+    else:
+        diff_args = ('--staged',)
     added_diff = cmd_output(
-        'git', 'diff', '--staged', '--diff-filter=A', '--raw',
+        'git', 'diff', '--diff-filter=A', '--raw', *diff_args, '--', *argv,
     )
```

Comment:
this shouldn't be needed -- use the arguments that are passed from pre-commit directly (you'll want to set up an `argparse.ArgumentParser` -- see how the rest of the tools do this)

Reply:
Initially I thought that too and wanted to rewrite it using `git status` instead of `git diff` to query mode. However, the hook is called "forbid **new** submodules" which apparently means that bumping somehow already existent submodules should be allowed. That's why we need `--diff-filter=A` and we need to query `PRE_COMMIT_*_REF` variables in order to ensure that no submodules have been added in any commits being validated, not only the last one.